### PR TITLE
Improve and add logging to run_equivocator_network test.

### LINKED ISSUE #1859: Non-det failure of `run_equivocator_network` test.
https://drone-auto-casper-network.casperlabs.io/casper-network/casper-node/1453/3/3 failed with

```
failures:

---- reactor::participating::tests::run_equivocator_network stdout ----
thread 'reactor::participating::tests::run_equivocator_network' panicked at 'assertion failed: `(left == right)`
  left: `[]`,
 right: `[PublicKey::Ed25519(5bc5542a73a3fd4e53c2f3c84838156cf8ce5de0db828a5644b1c6d883056aa6)]`', node/src/reactor/participating/tests.rs:319:13
```

source:
https://github.com/casper-network/casper-node/blob/dev/node/src/reactor/participating/tests.rs#L319

4.11.23 can we create equivocation through the diagnostic port?  Goal: make this deterministic and then address.   Use the equivocation mechanism available in Zug

### PR DESCRIPTION
**Pull request by afck**
The test is flaky on CI but always passes locally. This adds some logging so we can see whether it's a timing issue.

We also start isolating one of the two clone nodes for three rounds starting from the first _consensus_ message it sends now, instead of the first message it receives: If other messages are exchanged before the genesis timestamp (address gossiping?), the three round lengths could expire before the first consensus message is even sent.

See issue #1859.

### CODE REVIEW

File: node/src/reactor/participating/tests.rs [hunk 1]
```diff
@@ -325,19 +331,18 @@ async fn run_equivocator_network() {
...
-            ) {
-                return Either::Right(event);
-            }
             let now = Timestamp::now();
-            let first_message = if let Some(first_message) = maybe_first_message {
-                first_message
-            } else {
-                maybe_first_message = Some(now);
-                now
+            let first_message = match (&event, maybe_first_message) {
```
Comment (goral09) [inline node/src/reactor/participating/tests.rs]: It's first message's timestamp, right? Can we change the name?

**Review by goral09 — APPROVED** [review]

**Review by xcthulhu — APPROVED** [review]
LGTM

File: node/src/reactor/participating/tests.rs (hunk 1 above)
Comment (afck) [reply node/src/reactor/participating/tests.rs]: Done in https://github.com/casper-network/casper-node/pull/2133/commits/afaeaa465e49f1878bf902f5bdb415bd6b972df7.

**Comment by afck** [comment]
bors merge

**Comment by afck** [comment]
bors cancel

**Comment by afck** [comment]
bors merge

**Comment by afck** [comment]
bors merge

**Comment by afck** [comment]
bors merge

**Pull request merged**